source/extensions/filters/http/csrf/csrf_filter.cc

```diff
@@ -0,0 +1,114 @@
+#include "extensions/filters/http/csrf/csrf_filter.h"
+
+#include "envoy/stats/scope.h"
+
+#include "common/common/empty_string.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/utility.h"
+
+#include "extensions/filters/http/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Csrf {
+
+CsrfFilterConfig::CsrfFilterConfig(const envoy::config::filter::http::csrf::v2::CsrfPolicy& policy,
+                                   const std::string& stats_prefix, Stats::Scope& scope,
+                                   Runtime::Loader& runtime)
+    : stats_(generateStats(stats_prefix, scope)), policy_(generatePolicy(policy, runtime)) {}
+
+CsrfFilter::CsrfFilter(const CsrfFilterConfigSharedPtr config) : config_(config) {}
+
+Http::FilterHeadersStatus CsrfFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
+  determinePolicy();
+
+  if (!policy_->enabled() && !policy_->shadowEnabled()) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  if (!modifyMethod(headers)) {
+    return Http::FilterHeadersStatus::Continue;
+  }
+
+  bool isValid = true;
+  const auto& sourceOrigin = sourceOriginValue(headers);
```

Comment:
There's a bunch of places in this PR where your use of 'auto' makes the code harder to fully understand due to (a) the type being significant and (b) the declaration is not immediately in sight.

https://google.github.io/styleguide/cppguide.html#auto

There are about 10 in the PR I think, but I'll just leave this one comment.